File: rust/examples/secure_implementation/generate_key.rs
// you can generate a secure encryption key and salt using the cli
// or generate a secure encryption key and salt using the this example code
use base64::engine::general_purpose::STANDARD as BASE64_STANDARD;
use base64::Engine;
use skypydb::{EncryptionManager, Result};

fn main() -> Result<()> {
    let key = EncryptionManager::generate_key();
    let salt = EncryptionManager::generate_salt(32)?;

    println!("ENCRYPTION_KEY={key}");
    println!("SALT_KEY={}", BASE64_STANDARD.encode(salt));

    Ok(())
}

File: demo/examples/python/secure_implementation/generate_key.py
# you can generate a secure encryption key and salt using the cli
# or generate a secure encryption key and salt using the this example code

from skypydb.security import EncryptionManager

# Generate a secure encryption key
encryption_key = EncryptionManager.generate_key()
salt = EncryptionManager.generate_salt()
print(encryption_key) # don't show this key to anyone
print(salt) # don't show this salt to anyone
